src/core/thread/mle_router.cpp

```diff
@@ -50,7 +50,9 @@ namespace Mle {
 
 MleRouter::MleRouter(ThreadNetif &aThreadNetif):
     Mle(aThreadNetif),
-    mAdvertiseTimer(aThreadNetif.GetIp6().mTimerScheduler, &HandleAdvertiseTimer, this),
+    mRouterAdvertiseTimer(aThreadNetif.GetIp6().mTimerScheduler, TrickleTimer::kModeNormal,
+                          HandleRouterAdvertiseTimer, NULL, this),
+    mReedAdvertiseTimer(aThreadNetif.GetIp6().mTimerScheduler, &HandleReedAdvertiseTimer, this),
```

Comment:
Can we use a single timer object to handle both Router and REED roles?  In the case of REED, could we just stop/start the timer again in the callback?